catalog-rest-service/src/main/java/org/openmetadata/catalog/security/nativeauthorizer/NativeMetadataOperationAuthorizer.java

```diff
@@ -0,0 +1,63 @@
+package org.openmetadata.catalog.security.nativeauthorizer;
+
+import java.util.List;
+import org.jeasy.rules.api.Rule;
+import org.jeasy.rules.api.Rules;
+import org.jeasy.rules.api.RulesEngine;
+import org.jeasy.rules.api.RulesEngineParameters;
+import org.jeasy.rules.core.DefaultRulesEngine;
+import org.jeasy.rules.core.RuleBuilder;
+import org.openmetadata.catalog.entity.teams.User;
+import org.openmetadata.catalog.type.MetadataOperation;
+
+/**
+ * Authorizer for {@link MetadataOperation metadata operations} based on OpenMetadata's internal {@link
+ * org.openmetadata.catalog.entity.policies.Policy} format for AccessControl.
+ *
+ * <p>This authorizer uses {@link DefaultRulesEngine} provided by <a
+ * href="https://github.com/j-easy/easy-rules">j-easy/easy-rules</a> package.
+ *
+ * <p>The rules defined as {@link org.openmetadata.catalog.entity.policies.accessControl.Rule} are to be fetched from
+ * OpenMetadata's policy entity table and converted into {@link Rule} to be used within the {@link DefaultRulesEngine}
+ *
+ * <p>The facts are constructed based on 3 inputs for the Authorizer:
+ *
+ * <p>- {@link MetadataOperation operation} to be performed - {@link User} who performs the operation - {@link
+ * org.openmetadata.catalog.Entity} on which to operate on
+ */
+public class NativeMetadataOperationAuthorizer {
```

Comment:
we can rename this into DefaultAuthorizer https://github.com/open-metadata/OpenMetadata/blob/main/catalog-rest-service/src/main/java/org/openmetadata/catalog/security/DefaultCatalogAuthorizer.java

Reply:
Splitting out the PolicyEvaluator from the Authorizer (check component diagram in #1940)

DefaultCatalogAuthorizer will have an instance of PolicyEvaluator created during init() -- this change to come in future PRs.